src/main/java/seedu/cafectrl/data/Pantry.java

```diff
@@ -0,0 +1,119 @@
+package seedu.cafectrl.data;
+
+import seedu.cafectrl.data.dish.Ingredient;
+
+import java.util.ArrayList;
+
+/**
+ * Represents a pantry that stores ingredients and manages their quantities.
+ */
+public class Pantry {
+
+    private ArrayList<Ingredient> pantryStock;
+
+    /**
+     * Retrieves the current pantry stock from storage, which may include reading from a file (pantry.txt).
+     * @return An ArrayList of Ingredient objects representing the current pantry stock.
+     */
+    public ArrayList<Ingredient> getPantryStock() {
+        pantryStock.addAll(retrieveStockFromStorage());
+        return pantryStock;
+    }
+
+    /**
+     * Retrieves the pantry stock from storage, e.g., by reading from a file (pantry.txt).
+     * @return An ArrayList of Ingredient objects representing the pantry stock.
+     */
+    public ArrayList<Ingredient> retrieveStockFromStorage () {
+        ArrayList<Ingredient> pantryStock = new ArrayList<>();
+        //TODO: Add file reader to read from pantry.txt
+        return pantryStock;
+    }
+
+    /**
+     * Extracts the quantity (numeric part) from a given string containing both quantity and unit.
+     * @param qty A string containing both quantity and unit (e.g., "100g").
+     * @return An integer representing the extracted quantity.
+     */
+    public int extractQty(String qty) {
+        return Integer.parseInt(qty.replaceAll("[^0-9]", ""));
+    }
+
+    /**
+     * Extracts the unit (non-numeric part) from a given string containing both quantity and unit.
+     * @param qty A string containing both quantity and unit (e.g., "100g").
+     * @return A string representing the extracted unit.
+     */
+    public String extractUnit(String qty) {
+        return qty.replaceAll("[0-9]", "");
+    }
+
+    /**
+     * Adds or updates an ingredient in the pantry stock based on its name and quantity.
+     * @param name The name of the ingredient to add or update.
+     * @param qty The quantity of the ingredient (e.g., "100g").
+     * @return The Ingredient object that was added or updated in the pantry stock.
+     */
+    public Ingredient addIngredientToStock (String name, String qty) {
+        pantryStock = getPantryStock(); //get latest pantry stock
+        int ingredientIndex = -1;
+
+        //checks if ingredient exists in the pantry
+        ingredientIndex = getIndexOfIngredient(name);
+
+        //TODO: Add file writer to write update pantry.txt
+        return updateIngredient(name, qty, ingredientIndex);
+    }
```

Comment:
Since the main purpose of this method is to check whether the ingredient exists or not, perhaps you could create a different method called checkIngredientExists()

In your addIngredientToStock(), maybe you could do something like checkIngredientExist() ? updateIngredientQty() : create new ingredient and update the quantity